hw3/02-charts.js

```diff
@@ -0,0 +1,122 @@
+const backgroundColors = [
+  "rgba(54, 162, 235, 0.8)",
+  "rgba(255, 206, 86, 0.8)",
+  "rgba(255, 99, 132, 0.8)",
+  "rgba(75, 192, 192, 0.8)",
+  "rgba(153, 102, 255, 0.8)",
+  "rgba(255, 159, 64, 0.8)",
+  "rgba(199, 199, 199, 0.8)",
+  "rgba(83, 102, 255, 0.8)",
+  "rgba(40, 159, 64, 0.8)",
+  "rgba(210, 199, 199, 0.8)",
+  "rgba(78, 52, 199, 0.8)",
+];
+
+const borderColors = [
+  "rgba(54, 162, 235, 1)",
+  "rgba(255, 206, 86, 1)",
+  "rgba(255, 99, 132, 1)",
+  "rgba(75, 192, 192, 1)",
+  "rgba(153, 102, 255, 1)",
+  "rgba(255, 159, 64, 1)",
+  "rgba(159, 159, 159, 1)",
+  "rgba(83, 102, 255, 1)",
+  "rgba(40, 159, 64, 1)",
+  "rgba(210, 199, 199, 1)",
+  "rgba(78, 52, 199, 1)",
+];
+
+const Validator = {
+  Targaryen: ["Targaryan", "Targaryen"],
+  Tarly: ["Tarly", "Tarth"],
+  Tyrell: ["Tyrell"],
+  Stark: ["Stark"],
+  Baratheon: ["Baratheon"],
+  Lannister: ["Lannister", "Lanister"],
+  Greyjoy: ["Greyjoy"],
+  Clegane: ["Clegane"],
+  Baelish: ["Baelish"],
+  Seaworth: ["Seaworth"],
+  Unknown: ["Unknown", "None", "Unkown"],
+  "Free Folk": ["Free Folk"],
+  Naathi: ["Naathi"],
+  Bolton: ["Bolton"],
+  Naharis: ["Naharis"],
+  Lorathi: ["Lorathi", "Lorath"],
+  Mormont: ["Mormont"],
+  Sparrow: ["Sparrow"],
+  Viper: ["Viper"],
+  Sand: ["Sand"],
+  Worm: ["Worm"],
+  Qyburn: ["Qyburn"],
+  Bronn: ["Bronn"],
+};
+
+const Thronesurl = "https://thronesapi.com/api/v2/Characters";
+
+const fetchThrones = async () => {
+  try {
+    const response = await fetch(Thronesurl);
+    const data = await response.json();
+    return data;
+  } catch (error) {
+    console.error("Error:", error);
+    errormessage.style.display = "block";
+    return [];
+  }
+};
+
+const countCharactersByFamily = (characters) => {
+  const familyCounts = {};
+  characters.forEach((character) => {
+    const family = character.lastName ? character.lastName.trim() : "Unknown";
```

Comment:
I think in the data, there is a `family` attribute we can use

Reply:
ya both family and last name attribute seemed similar so i used last name